zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/GroupControllerTest.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.rest.controller;
+
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.camunda.identity.usermanagement.CamundaGroup;
+import io.camunda.identity.usermanagement.service.GroupService;
+import io.camunda.zeebe.gateway.protocol.rest.CamundaGroupRequest;
+import io.camunda.zeebe.gateway.protocol.rest.CamundaGroupResponse;
+import io.camunda.zeebe.gateway.protocol.rest.GroupSearchResponse;
+import io.camunda.zeebe.gateway.protocol.rest.SearchQueryRequest;
+import io.camunda.zeebe.gateway.rest.controller.usermanagement.GroupController;
+import java.net.URI;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ProblemDetail;
+
+@WebMvcTest(GroupController.class)
+public class GroupControllerTest extends RestControllerTest {
+
+  @MockBean private GroupService groupService;
+
+  @Test
+  void getGroupByIdShouldReturnGroup() {
+    final CamundaGroup camundaGroup = new CamundaGroup(1L, "demo");
+
+    final CamundaGroupResponse camundaGroupResponse = new CamundaGroupResponse();
+    camundaGroupResponse.setId(1L);
+    camundaGroupResponse.setName("demo");
+
+    when(groupService.findGroupById(1L)).thenReturn(camundaGroup);
+
+    webClient
+        .get()
+        .uri("/v2/groups/1")
+        .accept(MediaType.APPLICATION_JSON)
+        .exchange()
+        .expectStatus()
+        .is2xxSuccessful()
+        .expectBody(CamundaGroupResponse.class)
+        .isEqualTo(camundaGroupResponse);
+  }
+
+  @Test
+  void getGroupByIdThrowsExceptionWhenServiceThrowsException() {
```

Comment:
🔧 You could test the different exception types and that they lead to different status codes (400 and 500).